airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -115,8 +115,8 @@ export const DEFAULT_BUILDER_STREAM_VALUES: Omit<BuilderStream, "id"> = {
   },
 };
 
-function getInferredInputList(values: BuilderFormValues): BuilderFormInput[] {
-  if (values.global.authenticator.type === "ApiKeyAuthenticator") {
+function getInferredInputList(global: BuilderFormValues["global"]): BuilderFormInput[] {
```

Comment:
Not taking the full `BuilderFormValues`, but just the required parts allows for memoization (otherwise every change will require to re-run this function)